server/webhook.go

```diff
@@ -368,40 +364,73 @@ func (s *Server) processFile(ctx context.Context, pushEvent vcs.PushEvent, repo
-	} else {
-		// This is a migration-based DDL or DML file and we would allow it for both DDL and SDL schema change type project.
-		// For DDL schema change type project, this is expected.
-		// For SDL schema change type project, we allow it because:
-		// 1) DML is always migration-based.
-		// 2) We may have a limitation in SDL implementation.
-		// 3) User just wants to break the glass.
-		migrationDescription = migrationInfo.Description
-		migrationDetailList, activityCreateList = s.prepareIssueFromPushEventDDL(ctx, repo, pushEvent, file, fileType, migrationInfo)
+	var createdIssueList []string
+	// Default to DATA migration. If later we found any MIGRATE migration file, we set migrationType to MIGRATE.
+	migrationTypeDDL := db.Data
+
+	// There are possibly multiple files in the push event.
+	// Each file corresponds to a (database name, schema version) pair.
+	// We want the migration statements are sorted by the file's schema version, and grouped by the database name.
+	fileInfoListSorted := sortFilesBySchemaVersionGroupByDatabase(fileInfoList)
+	for _, fileInfo := range fileInfoListSorted {
+		if fileInfo.fType == schemaFileType {
+			if repo.Project.SchemaChangeType == api.ProjectSchemaChangeTypeSDL {
+				// Create one issue per schema file for SDL project.
+				migrationDetailListForFile, activityCreateListForFile := s.prepareIssueFromPushEventSDL(ctx, repo, pushEvent, fileInfo.migrationInfo, fileInfo.item.FileName)
+				activityCreateList = append(activityCreateList, activityCreateListForFile...)
+				if len(migrationDetailListForFile) != 0 {
+					issueName := fmt.Sprintf("Apply schema diff by file %s", strings.TrimPrefix(fileInfo.item.FileName, repo.BaseDirectory+"/"))
+					creatorID := s.getIssueCreatorID(ctx, pushEvent.FileCommit.AuthorEmail)
+					if err := s.createIssueFromMigrationDetailList(ctx, issueName, pushEvent, creatorID, repo.ProjectID, fileInfo.migrationInfo.Type, migrationDetailListForFile); err != nil {
+						return "", false, activityCreateList, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create issue").SetInternal(err)
+					}
+					createdIssueList = append(createdIssueList, issueName)
+				}
+			} else {
+				log.Debug("Ignored schema file for non-SDL project", zap.String("fileName", fileInfo.item.FileName), zap.String("type", string(fileInfo.item.ItemType)))
+			}
+		} else { // fileInfo.fType == migrationFileType
+			// This is a migration-based DDL or DML file and we would allow it for both DDL and SDL schema change type project.
+			// For DDL schema change type project, this is expected.
+			// For SDL schema change type project, we allow it because:
+			// 1) DML is always migration-based.
+			// 2) We may have a limitation in SDL implementation.
+			// 3) User just wants to break the glass.
+			if fileInfo.migrationInfo.Type == db.Migrate {
+				migrationTypeDDL = db.Migrate
+			}
```

Comment:
We only set to `db.Migrate` but never sets it back to `db.Data`, what happens if the first file is DDL but the next file is DML? Both of them will be treated as DDL.

Reply:
Yes, and this is by design.

IMO, if uses merge DDL with DML scripts, it is highly possible that they are creating a new table and inserting some seed data.

Reply:
>it is highly possible that they are creating a new table and inserting some seed data.

Then it is not possible to do pure DML afterward in a single push? This implication doesn't sound reasonable to be imposed.

Reply:
If a "pure DML" is intended, the user can create a different PR/MR for it.

The code here is dealing with the case that the user merges DML/DDL in a single PR/MR.

Reply:
OK, can you add a code comment about "our design decision"?

Reply:
Sure. Updated in L287-288.